src/Checkbox/index.js

```diff
@@ -9,67 +9,115 @@ import { Icon } from './Icon'
-            checked={checked}
-            disabled={disabled}
-            valid={valid}
-            error={error}
-            warning={warning}
-            indeterminate={indeterminate}
-        />
-
-        <Label required={required}>{label}</Label>
-
-        <style jsx>{`
-            label {
-                display: flex;
-                flex-direction: row;
-                align-items: center;
-                justify-content: flex-start;
-                cursor: pointer;
-                pointer-events: all;
-                user-select: none;
-            }
-
-            .disabled {
-                cursor: not-allowed;
-                color: ${theme.disabled};
-            }
-        `}</style>
-    </label>
-)
+            valid,
+            warning,
+            error,
+            focus = this.state.focus,
+        } = this.props
+
+        return (
+            <Fragment>
+                <label
+                    className={cx('base', className, {
+                        disabled,
+                        focus,
+                    })}
+                >
+                    <Input
+                        name={name}
+                        value={value}
+                        checked={checked}
+                        disabled={disabled}
+                        focus={focus}
+                        onChange={onChange}
+                        onFocus={this.onFocus}
+                        onBlur={this.onBlur}
+                    />
+
+                    <Icon
+                        focus={focus}
+                        checked={checked}
+                        disabled={disabled}
+                        valid={valid}
+                        error={error}
+                        warning={warning}
+                        indeterminate={indeterminate}
+                    />
+
+                    <Label required={required}>{label}</Label>
+
+                    <style jsx>{`
+                        label {
+                            display: flex;
+                            flex-direction: row;
+                            align-items: center;
+                            justify-content: flex-start;
+                            cursor: pointer;
+                            pointer-events: all;
+                            user-select: none;
+                        }
+
+                        .disabled {
+                            cursor: not-allowed;
+                            color: ${theme.disabled};
+                        }
+
+                        .focus {
+                            outline: 1px;
+                        }
+                    `}</style>
+                </label>
+            </Fragment>
+        )
+    }
+}
 
 Checkbox.propTypes = {
     onChange: propTypes.func.isRequired,
 
+    value: propTypes.string.isRequired,
     name: propTypes.string.isRequired,
     label: propTypes.string.isRequired,
 
+    onFocus: propTypes.func,
+    onBlur: propTypes.func,
```

Comment:
Maybe we should provide defaults to these to `null` so that React doesn't throw propType warnings when we omit them and they get sent as `undefined` to the actual `input`?

Reply:
Didn't notice that.. I didn't see an error in my console, but makes perfect sense.
Null will cause the same problem, so I just removed the `isRequired` from the internal input components